extensions/amp-yotpo/0.1/amp-yotpo.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getIframe} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export class AmpYotpo extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    // We are conservative about loading heavy embeds.
+    // This will still start loading before they become visible, but it
+    // won't typically load a large number of embeds.
+    return 0.75;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+  }
+
+  layoutCallback() {
+    user().assert(this.element.getAttribute('data-app-key'),
+        'The data-app-key attribute is required for <amp-yotpo> %s',
+        this.element);
+    user().assert(this.element.getAttribute('data-widget-type'),
+        'The data-widget-type attribute is required for <amp-yotpo> %s',
+        this.element);
+    const iframe = getIframe(this.win, this.element, 'yotpo');
```

Comment:
Does your component need to resize automatically? If the page author can not reliably know the size or at least the  aspect ratio of the embedded content, you may want your 3p script (yopto.js) to send a updateDimention request and have this handled here. For reference see how `amp-gist` handles resizing: https://github.com/ampproject/amphtml/blob/master/extensions/amp-gist/0.1/amp-gist.js#L64 and https://github.com/ampproject/amphtml/blob/master/3p/github.js#L66

Reply:
Done